### DISCUSSION TITLE
How to wire Adafruit Itsybitsy RP2040

### DISCUSSION THREAD

**Discussion by tommyguuuun** [awawa-dev/HyperSerialPico#18] [General]
Hey there,
as i have connection issues with my current WLED setup, i ordered an Adafruit Itsybitsy RP2040 for using HyperSerialPico.

As i am not that experienced, i have a question how to wire it correctly:

Currently my ESP32 is powered by the same power supply that is powering my led strip (just using wago clamps to split it). But as the Adafruit is powered by the USB connection, that will come from my Raspi 4, i'm wondering how to wire the ground? Can i use the ground coming from the powersupply, or does it have to be the one from the Raspi 4?

Here is a picture how i wired the ground and data to the board:

![IMG_5727](https://github.com/awawa-dev/HyperSerialPico/assets/26925125/e6f6187e-2566-4cd5-812c-8be156250beb)

Thanks in advance!

**Comment by awawa-dev** [comment]
Hi
The black cable connected to the ground of the Adafruit Itsybitsy (G) should be connected to the ground of the LED strip to create a common ground (first component). The green cable should be connected to LED strip data.

USB will provide: ground (second component of common ground), +5V and USB Adalight data provider for Itsybitsy.

Never connect +5V from the LED power supply to Pico/ESP board (Adafruit Itsybitsy in this case) if this board is already receiving +5V from USB (Rpi4).

**Reply by tommyguuuun** [reply]
Perfect, thanks so much for help! And thanks for the great work with HyperHDR!
